src/query-helpers.js

```diff
@@ -49,10 +50,27 @@ function makeSingleQuery(allQuery, getMultipleError) {
         container,
       )
     }
-    return els[0] || null
+    const element = els[0] || null
+
+    if (getConfig().showSuggestions) {
+      const suggestion = getSuggestedQuery(element)
+      if (suggestion && !allQuery.name.endsWith(suggestion.queryName)) {
+        throw getSuggestionError(suggestion.toString(), container)
+      }
+    }
+    return element
```

Comment:
Copy from [Twitter](https://twitter.com/meijer_s/status/1265217798298578944?s=20):

I think it's awesome to have suggestions embedded within this project. But as expressed before, I'm still concerned about the developer experience.

It's currently throwing errors when there are slightly better queries available. I'm afraid this would be the first thing I would disable.

Perhaps it's better to log warnings instead? That's also what the config name implies `showSuggestions`.

If we're throwing, a config property like `strict` would be a better match IMO.

Reply:
@smeijer yeah the issue is pointing people exactly to the line we are suggesting them to change.  I'm open to other ideas but throwing an error in jest will take them directly to where the suggestion is pointing. Think of this In the same way that detect leaks works in jest. It only breaks tests when you turn it on.  That's why this is opt-in and disabled by default.

Reply:
I'm good with renaming the config property but `strict` feels a bit too vague to me. @kentcdodds  any thoughts here?

Reply:
How would the stacks look with `console.warn`?

Reply:
> It's currently throwing errors when there are slightly better queries available.

Actually on many teams people go straight to getByTestId and I would argue it's not just slightly better but WAY better to use getByRole so for me I actually want this to throw in those cases

Reply:
I can try it out @eps1lon  and show the differences here

Reply:
@eps1lon unfortunately warn didn't work :(